docs/types/value-types.rst

```diff
@@ -188,25 +188,22 @@ Type conversions:
 Implicit conversions from ``address payable`` to ``address`` are allowed, whereas conversions from ``address`` to ``address payable``
 must be explicit via ``payable(<address>)``.
 
-:ref:`Address literals<address_literals>` can be implicitly converted to ``address payable``.
-
 Explicit conversions to and from ``address`` are allowed for integers, integer literals, ``bytes20`` and contract types with the following
 caveat:
 The result of a conversion of the form ``address(x)``
 has the type ``address payable``, if ``x`` is of integer or fixed bytes type,
-a literal or a contract with a receive or payable fallback function.
+or a contract with a receive or payable fallback function.
 If ``x`` is a contract without a receive or payable fallback function,
 then ``address(x)`` will be of type ``address``.
+Similarly, if ``x`` is a literal, then ``address(x)`` will also be of type ``address``.
 In external function signatures ``address`` is used for both the ``address`` and the ``address payable`` type.
 
 Only expressions of type ``address`` can be converted to type ``address payable`` via ``payable(<address>)``.
 
 .. note::
-    It might very well be that you do not need to care about the distinction between ``address``
-    and ``address payable`` and just use ``address`` everywhere. For example,
-    if you are using the :ref:`withdrawal pattern<withdrawal_pattern>`, you can (and should) store the
-    address itself as ``address``, because you invoke the ``transfer`` function on
-    ``msg.sender``, which is an ``address payable``.
+    If you store a variable of ``address`` type and plan to send Ether to it, then
+    store it as ``address payable`` to make this requirement visible. Also try to pull this
+    type as far out as possible or do the conversion as early as possible.
```

Comment:
Not sure pulling the type far out is that easy to understand... do you basically mean "Prefer the type ``address payable``, whenever you can be sure that an ``address`` is in fact ``payable``" or something like that?